crucible-go/src/Lang/Crucible/Go/Overrides.hs

```diff
@@ -0,0 +1,169 @@
+  liftIO $ addAssumption sym (LabeledPred (regValue b) $
+                              AssumptionReason loc "assume")
+  return Ctx.empty
+
+do_assert :: IsSymInterface sym
+          => C.OverrideSim p sym ext gret
+          (EmptyCtx ::> StringType Unicode ::> StringType Unicode ::> BoolType)
+          (StructType EmptyCtx) (RegValue sym (StructType EmptyCtx))
+do_assert = do
+  sym <- C.getSymInterface
+  RegMap (Empty :> mgs :> file :> b) <- C.getOverrideArgs
+  loc <- liftIO $ W4.getCurrentProgramLoc sym
+  liftIO $ addDurableAssertion sym (LabeledPred (regValue b)
+                                    (C.SimError loc $ C.AssertFailureSimError
+                                     "assertion_failure" ""))
+  return Ctx.empty
+
+go_overrides :: (IsSymInterface sym, 1 <= w)
+             => NatRepr w
+             -> [(SomeOverride (Crux.Model sym) sym ext)]
+go_overrides w =
+  [ mkSomeOverride "crucible" "FreshInt" Ctx.empty (BVRepr w) (fresh_int w)
+  , mkSomeOverride "crucible" "FreshInt8" Ctx.empty
+    (BVRepr (knownNat :: NatRepr 8)) fresh_int'
+  , mkSomeOverride "crucible" "FreshInt16" Ctx.empty
+    (BVRepr (knownNat :: NatRepr 16)) fresh_int'
+  , mkSomeOverride "crucible" "FreshInt32" Ctx.empty
+    (BVRepr (knownNat :: NatRepr 32)) fresh_int'
+  , mkSomeOverride "crucible" "FreshInt64" Ctx.empty
+    (BVRepr (knownNat :: NatRepr 64)) fresh_int'
+  , mkSomeOverride "crucible" "FreshUint" Ctx.empty (BVRepr w) (fresh_int w)
+  , mkSomeOverride "crucible" "FreshUint8" Ctx.empty
+    (BVRepr (knownNat :: NatRepr 8)) fresh_int'
+  , mkSomeOverride "crucible" "FreshUint16" Ctx.empty
+    (BVRepr (knownNat :: NatRepr 16)) fresh_int'
+  , mkSomeOverride "crucible" "FreshUint32" Ctx.empty
+    (BVRepr (knownNat :: NatRepr 32)) fresh_int'
+  , mkSomeOverride "crucible" "FreshUint64" Ctx.empty
+    (BVRepr (knownNat :: NatRepr 64)) fresh_int'
+  , mkSomeOverride "crucible" "FreshString" Ctx.empty
+    (StringRepr UnicodeRepr) $ fresh_string UnicodeRepr
+  , mkSomeOverride "crucible" "Assume"
+    (Ctx.Empty :> StringRepr UnicodeRepr :> StringRepr UnicodeRepr :> BoolRepr)
+    (StructRepr Ctx.empty) do_assume
+  , mkSomeOverride "crucible" "Assert"
+    (Ctx.Empty :> StringRepr UnicodeRepr :> StringRepr UnicodeRepr :> BoolRepr)
+    (StructRepr Ctx.empty) do_assert ]
+
+lookupOverride :: Text
+               -> FunctionName
+               -> [SomeOverride p sym ext]
+               -> Maybe (SomeOverride p sym ext)
+lookupOverride _pkgName _fName [] = Nothing
+lookupOverride pkgName fName
+  (o@(SomeOverride pkg _argsRepr _retRepr override) : overrides) =
+  if pkgName == pkg && fName == overrideName override then
+    Just o
+  else
+    lookupOverride pkgName fName overrides
+
+lookupOverride' :: Maybe (Text, FunctionName)
+                -> [SomeOverride p sym ext]
+                -> Maybe (SomeOverride p sym ext)
+lookupOverride' nm overrides =
+  nm >>= flip (uncurry lookupOverride) overrides
```

Comment:
Is it worth a note to make these into a `Map` or `HashMap` one day?

Reply:
Good idea.